samples/repo-search/android-views/src/main/java/com/example/redwood/reposearch/android/views/RealHostApi.kt

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright (C) 2023 Square, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.example.redwood.reposearch.android.views
+
+import com.example.redwood.reposearch.treehouse.HostApi
+import kotlinx.coroutines.suspendCancellableCoroutine
+import okhttp3.Call
+import okhttp3.Callback
+import okhttp3.Headers.Companion.toHeaders
+import okhttp3.OkHttpClient
+import okhttp3.Request
+import okhttp3.Response
+import okio.IOException
+
+class RealHostApi(
+  private val client: OkHttpClient,
+) : HostApi {
+  override suspend fun httpCall(url: String, headers: Map<String, String>): String {
+    return suspendCancellableCoroutine { continuation ->
```

Comment:
You're not actually using cancelation here. Needs something like `continuation.invokeOnCancelation { call.cancel() }` or whatever the API is...

Reply:
Tbh I'd copy-pasted this from the [Emoji Search sample](https://github.com/cashapp/redwood/blob/trunk/samples/emoji-search/android-views/src/main/java/com/example/redwood/emojisearch/android/views/RealHostApi.kt). Whatever fixes are done here, they should probably also be done in Emoji Search. Do you think it's worth creating a separate HTTP module within the samples directory in order to avoid duplication? Usually I'd think it best, but as it's  a sample directory, I don't want to overcomplicate things and detract from the stuff that matters (i.e., the Treehouse stuff).

Reply:
Haha 😬 

Uhhh I'm not sure what we should do to be honest. I definitely don't want to go overkill on de-duplication because of the sample nature as you said.

I guess we can clean both of them up in a follow-up and then I can try and resume work on `fetch` in Zipline since that's the final solution. I have it started here: https://github.com/cashapp/zipline/compare/trunk...jw/fetch/2022-12-24.